test/fixtures/source-map/fake-source-map-url.js

```diff
@@ -0,0 +1,7 @@
+const a = 99;
+if (true) {
+  const b = 101;
+} else {
+  const c = 102;
+}
+const sm = '//# sourceMappingURL=https://ci.nodejs.org/402'
```

Comment:
this is technically an unsolvable problem, assuming an adversary opponent. V8 has APIs for this on UnboundScript, can we use those instead?

Reply:
V8 also just looks for comments of the form `//[#@]\s<name>=\s*<value>\s*.*`…

Reply:
For reference, here's a [helper in the v8 inspector ](https://github.com/v8/v8/blob/master/src/inspector/search-util.cc#L25). I believe it might be susceptible to the same bug.

Should we just leave this as a known issue with source maps?

Reply:
What I mean is, UnboundScript::GetSourceMappingURL uses information from the actual parser, it knows the difference between an actual comment and a tricky string.

Reply:
Looking into this further, the helper in the V8 inspector is only used if parsing the script failed.

Reply:
@devsnek I think this would be a pretty significant refactor, and we'd need a similar fallback for when parsing fails.

I like the idea that in some cases we could just use v8's source map extraction, mind you.

Reply:
yeah not blocking or anything, but I think it would be much nicer.

Reply:
I'm on the fence with this PR, just because I fear the complex regex slowing down module loading ... perhaps before we land a fix like this I could dig in to your suggestion.